### PR TITLE
Final Form -> Formik

### PR DESCRIPTION
**Pull request by NicholasLYang**
Converted from Final Form to Formik. While I really wanted to like Final Form, Formik ultimately plays nicer with TypeScript. The transition was pretty straightforward, since both libraries have very similar APIs.

There's a bit of general clean up in this PR too. Hopefully this will put us in good shape for next year.

### CODE REVIEW

**Comment by themichaelyang** [comment]
Resolved #146

File: src/js/modules/constants.ts [hunk 1]
```diff
@@ -179,3 +180,50 @@ export const SPONSORS_INFO: Array<SponsorFields> = [
     square: false
   }
 ];
+
+export const emptyApplyFormData = {
```
Comment (themichaelyang) [inline src/js/modules/constants.ts]: Does this need to be ApplyFormData?

Comment (NicholasLYang) [reply src/js/modules/constants.ts]: As in type annotation? I think TS infers it just fine

**Comment by themichaelyang** [comment]
Getting this bug when I click on "Edit Application". Any ideas?

Screen Shot 2019-06-09 at 3 59 07 PM

File: src/js/modules/forms/components/ConfirmationForm.tsx [hunk 2]
```diff
@@ -96,137 +94,131 @@ const ConfirmationForm: React.FunctionComponent<Props> | undefined = ({
...
-                        </Radio>
-                      </label>
-                      <Condition when="location" isNot="cannot-attend">
-                        <div className={classes.resumeUpload}>
-                          <UploadResumeButton
-                            uid={user.uid}
-                            label="Please upload your resume as a PDF, so we can share it
+    return <div> Redirecting to login...</div>;
+  }
+  if (!IS_CONFIRMATION_OPEN) {
```
Comment (themichaelyang) [inline src/js/modules/forms/components/ConfirmationForm.tsx]: good stuff

**Pull request merged**